elephant/spectral.py

```diff
@@ -523,6 +529,543 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
+        elif data.shape[axis] < n_segments:
+            raise ValueError("n_segments must be smaller than the data length")
+        # when only *n_segments* is given, *n_per_seg* is determined by solving
+        # the following equation:
+        #  n_segments * n_per_seg - (n_segments-1) * overlap * n_per_seg =
+        #     data.shape[-1]
+        #  --------------------   ===============================  ^^^^^^^^^^^
+        # summed segment lengths        total overlap              data length
+        n_per_seg = int(data.shape[axis] /
+                        (n_segments - overlap * (n_segments - 1)))
+
+    n_overlap = int(n_per_seg * overlap)
+    n_overlap_step = n_per_seg - n_overlap
+    n_segments = int((length_signal - n_overlap) / (n_per_seg - n_overlap))
+
+    # Generate frequencies for spectral measure estimate
+    if func_params_dict.get('return_onesided'):
+        freqs = np.fft.rfftfreq(n_per_seg, d=1/fs)
+    else:
+        freqs = np.fft.fftfreq(n_per_seg, d=1/fs)
+
+    # Zero-pad signal to fit segment length
+    remainder = length_signal % n_overlap_step
+
+    data = np.pad(data, [(0, 0), (0, remainder)],
+                  mode='constant', constant_values=0)
+
+    # Generate array for storing cross spectra estimates of segments
+    seg_estimates = np.zeros((n_segments,
+                              data.shape[0],
+                              data.shape[0],
+                              len(freqs)),
+                             dtype=np.complex64)
+
+    n_overlap_step = n_per_seg - n_overlap
+
+    for i in range(n_segments):
+
+        _, estimate = func(
+            data[:, i * n_overlap_step:i * n_overlap_step + n_per_seg],
+            **func_params_dict)
+
+        seg_estimates[i] = estimate
+
+    avg_estimate = np.mean(seg_estimates, axis=0)
+
+    return freqs, avg_estimate
+
+
+def segmented_multitaper_cross_spectrum(signals, n_segments=1,
+                                        len_segment=None,
+                                        frequency_resolution=None, overlap=0.5,
+                                        fs=1, nw=4, num_tapers=None,
+                                        peak_resolution=None,
+                                        return_onesided=True):
+    """
+    Estimates the cross spectrum of a given 'neo.AnalogSignal' using the
+    Multitaper method on segments of the data.
+
+    The cross spectrum is obtained through the following steps
+
+    1. Cut the given data into several overlapping segments. The degree of
+       overlap can be specified by parameter `overlap` (default is 0.5,
+       i.e. segments are overlapped by the half of their length).
+       The number and the length of the segments are determined according
+       to the parameters `n_segments`, `len_segment` or `frequency_resolution`.
+       By default, the data is not cut into  segments;
+
+    2. Calculate 'num_tapers' approximately independent estimates of the
```

Comment:
For the segmented functions, the documentation should not repeat what belongs to the single-segment function. As this function segments the data (according to 1), loops over the segments computing the cross-spectrum, and obtains an average at the end, this docstring should worry about describing this algorithm only. Any specifics about the cross-spectrum computation are expected to be part of the single-segment function, and then a `:func:` reference should provide the link (e.g., for each segment, obtain the cross spectrum using :func:\`multitaper_cross_spectrum\`). A `See Also` section should also be added with the reference to the single segment function. This avoids redundancy. However (if there are any) here you should describe any specific requirements for applying the single segment function to each segment during this segmented computation. The objective is to make clear to the user what the segmented estimate is, not the details about the cross spectrum using multitaper.

Reply:
We would like to keep it as is